components/bandwidth/bandwidth.app.js

```diff
@@ -1,4 +1,59 @@
+const bandwidthMessaging = require("@bandwidth/messaging");
+
 module.exports = {
   type: "app",
   app: "bandwidth",
+  propDefinitions: {
+    messageTo: {
+      type: "string",
+      label: "To",
+      description: "The number the message will be sent to, in E164 format ex `+19195551234`.",
+    },
+    from: {
+      type: "string",
+      label: "From",
+      description: "The number the call or message event will come from, in E164 format ex `+19195551234`.",
+    },
+    message: {
+      type: "string",
+      label: "Message",
+      description: "The text message content.",
+    },
+    messagingApplicationId: {
+      type: "string",
+      label: "Messaging Application ID",
+      description: "The ID from the messaging application created in the [Bandwidth Dashboard](https://dashboard.bandwidth.com).\n\nThe application must be associated with the location that the `from` number lives on.",
+    },
+    mediaUrl: {
+      type: "string[]",
+      label: "Media URL",
+      description: "Publicly addressible URL of the media you would like to send with the SMS.",
+    },
+  },
+  methods: {
+    getMessagingClient() {
+      return new bandwidthMessaging.Client({
+        basicAuthUserName: this.$auth.username,
+        basicAuthPassword: this.$auth.password,
+      });
+    },
+    async sendSms(to, from, message, messagingApplicationId) {
+      const controller = new bandwidthMessaging.ApiController(this.getMessagingClient());
+      const data = {
+        applicationId: messagingApplicationId,
+        to: [
+          to,
+        ],
+        from: from,
+        text: message,
+      };
+      try {
+        const response = await controller.createMessage(this.$auth.accountId, data);
+        return response;
+      } catch (error) {
+        console.log(error.result);
```

Comment:
When an action returns an error in a step, Pipedream should display the full error, with the stack trace, just below the step. So you can remove the `try / catch` logic and simply let the error bubble up if one occurs.

There are cases where a user may want to wrap a step with try / catch logic on their own (which we're planning to allow on a step-level), but for now, our default behavior has been to let the error bubble up and halt the workflow, which will happen anytime a step throws an error. That way the user will be notified of the error in [the global error workflow](https://pipedream.com/docs/workflows/error-handling/global-error-workflow/#global-error-workflow).

Reply:
ah - here are those error docs 😅